contrib/babelfishpg_tsql/src/tsqlIface.cpp

```diff
@@ -3393,6 +3471,76 @@ static void process_query_specification(
 					mutator->add(column_alias_as->start->getStartIndex(), "", " AS ");
 			}
 		}
+		else if(process_local_id_assignement && elem->LOCAL_ID() && elem->EQUAL())
+		{
+			std::string var_str = ::getFullText(elem->LOCAL_ID());
+			PLtsql_nsitem *nse = pltsql_ns_lookup(pltsql_ns_top(), false, var_str.c_str(), nullptr, nullptr, nullptr);
+			char *repl_text = NULL;
+
+			Assert(elem->expression());
+			if (!nse)
+				throw PGErrorWrapperException(ERROR, ERRCODE_SYNTAX_ERROR, format_errmsg("\"%s\" is not a known variable", var_str.c_str()), getLineAndPos(elem));
+
+			/* Rewrite @var = expr to @var=sys.pltsql_assign_var(dno, cast((expr) as type)) */
+			repl_text = psprintf("sys.pltsql_assign_var(%d, %s)",
+									nse->itemno,
+									rewrite_assignment_expression((PLtsql_var *) pltsql_Datums[nse->itemno],
+																	elem->expression()));
+
+			handle_local_ids_for_expression(elem->expression());
+			mutator->add(elem->expression()->start->getStartIndex(), ::getFullText(elem->expression()), std::string(repl_text));
```

Comment:
local_id_positions is getting init inside enter_Local_id which registers all positions of local_id within tsql batch which will be re-written using quotes later. We don't want to do it for assignment expression and so we are just removing positions of local_ids that are part of given expression to avoid re-writing.